src/devrules/cli_commands/branch.py

```
@@ -333,29 +356,89 @@ def delete_merged(
+                subheader="Branches already merged and owned by you:",
+                messages=[f"{idx}. {b}" for idx, b in enumerate(final_candidates, 1)],
+            )
+            typer.echo()
+            choices = typer.prompt("Enter number, multiple separated by a space", type=str)
+            choices = choices.split(" ")
+            delete_branches = []
+            try:
+                choices = [int(choice) for choice in choices]
+            except ValueError:
+                typer.secho(msg.INVALID_CHOICE, fg=typer.colors.RED)
+                raise typer.Exit(code=1)
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)